controllers/auth_controller.go

```diff
@@ -52,3 +62,53 @@ func (ctrl *authController) Register(context *gin.Context) {
 
 	context.JSON(http.StatusOK, gin.H{"message": "Success", "user": body})
 }
+
+func (ctrl *authController) Login(context *gin.Context) {
+	type ILogin struct {
+		EmailAddress string `json:"emailAddress"`
+		Username string `json:"username"`
+		Password     string `json:"password" binding:"required"`
+	}
+	var body ILogin
+	if err := context.ShouldBindJSON(&body); err != nil {
+		context.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	// Determine if user is logging in with email or username
+	var loginType string
+	var loginValue string
+	if body.EmailAddress != "" {
+		loginType = "emailAddress"
+		loginValue = body.EmailAddress
+	} else if body.Username != "" {
+		loginType = "username"
+		loginValue = body.Username
+	} else {
+		context.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Neither username/email address not provided."})
```

Comment:
nit: the error message is a double negative and has the wrong meaning. Change to "Neither username nor email were provided"